### PR TITLE
Change bindgen version from 0.59.2 to 0.57.

### PR DESCRIPTION
**Pull request by egrimley-arm**
The toolchain version currently used for the IceCap build does not
accept the code generated by bindgen 0.59.2.

### CODE REVIEW

**Comment by egrimley-arm** [comment]
I'm now not sure that this change is required. I couldn't reproduce the failure after `make clean` and `rm */Cargo.lock` just now, though I thought I had before. Is there any state that survives through  `make clean` and `rm */Cargo.lock`?

**Comment by dominic-mulligan-arm** [comment]
I don't think `rm */Cargo.lock` is sufficient to remove all state, as we have deeply nested directory trees.  You should use `make clean-cargo-lock` which removes all lock files instead.

Nevertheless, I think there's no harm in merging this, even if it apparently is no longer strictly needed, right?

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
Let's wait for a second opinion, though, before merging this.  @mathias-arm are you happy with the version bump?

**Review by ShaleXIONG — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
Second opinion obtained.  +1+1 = +1 reached, merging.

**Comment by egrimley-arm** [comment]
> I don't think `rm */Cargo.lock` is sufficient to remove all state, as we have deeply nested directory trees. You should use `make clean-cargo-lock` which removes all lock files instead.

`make clean-cargo-lock` seems to just do `rm -f */Cargo.lock`:

https://github.com/veracruz-project/veracruz/blob/59864749459683d240b617d7396b653a0200b157/workspaces/Makefile#L56

**Comment by dominic-mulligan-arm** [comment]
Ah, I was thinking of the old build system where the top-level `Makefile` used to recursively call `clean-cargo-lock` for various system components.  With the move to workspaces, it seems the situation has become simpler...

**Pull request merged**
